### ISSUE TITLE
[Bug]: OpenAI Assistant Agent Error Code 400

### ISSUE THREAD

**Issue by carterrees-entrata** [run-llama/llama_index#8873]
### Bug Description

I have been running the OpenAI Assistant agents from this page on [docs page](https://docs.llamaindex.ai/en/stable/examples/agent/openai_assistant_agent.html#assistant-with-query-engine-tools). Simple Agent (no external tools) and Assistant with Built-in Retrieval work fine. However, when I run Assistant with Query Engine tools I get the error shown below. 

### Version

0.8.86

### Steps to Reproduce

I run each cell as shown in the docs page [here](https://docs.llamaindex.ai/en/stable/examples/agent/openai_assistant_agent.html#assistant-with-query-engine-tools).

### Relevant Logs/Tracbacks

```shell
---------------------------------------------------------------------------
BadRequestError                           Traceback (most recent call last)
Cell In[59], line 1
----> 1 agent = OpenAIAssistantAgent.from_new(
      2     name="SEC Analyst",
      3     instructions="You are a QA assistant designed to analyze sec filings.",
      4     tools=query_engine_tools,
      5     instructions_prefix="Please address the user as Jerry.",
      6     verbose=True,
      7     run_retrieve_sleep_time=1.0,
      8 )
      9 response = agent.chat("What was Lyft's revenue growth in 2021?")

File ~/opt/miniconda3/envs/llm/lib/python3.9/site-packages/llama_index/agent/openai_assistant_agent.py:190, in OpenAIAssistantAgent.from_new(cls, name, instructions, tools, openai_tools, thread_id, model, instructions_prefix, run_retrieve_sleep_time, files, callback_manager, verbose)
    188 # TODO: openai's typing is a bit sus
    189 all_openai_tools = cast(List[Any], all_openai_tools)
--> 190 assistant = client.beta.assistants.create(
    191     name=name,
    192     instructions=instructions,
    193     tools=cast(List[Any], all_openai_tools),
    194     model=model,
    195     file_ids=list(file_dict.keys()),
    196 )
    197 return cls(
    198     client,
    199     assistant,
   (...)
    204     verbose=verbose,
    205 )

File ~/opt/miniconda3/envs/llm/lib/python3.9/site-packages/openai/resources/beta/assistants/assistants.py:95, in Assistants.create(self, model, description, file_ids, instructions, metadata, name, tools, extra_headers, extra_query, extra_body, timeout)
     57 """
     58 Create an assistant with a model and instructions.
     59 
   (...)
     92   timeout: Override the client-level default timeout for this request, in seconds
     93 """
     94 extra_headers = {"OpenAI-Beta": "assistants=v1", **(extra_headers or {})}
---> 95 return self._post(
     96     "/assistants",
     97     body=maybe_transform(
     98         {
     99             "model": model,
    100             "description": description,
    101             "file_ids": file_ids,
    102             "instructions": instructions,
    103             "metadata": metadata,
    104             "name": name,
    105             "tools": tools,
    106         },
    107         assistant_create_params.AssistantCreateParams,
    108     ),
    109     options=make_request_options(
    110         extra_headers=extra_headers, extra_query=extra_query, extra_body=extra_body, timeout=timeout
    111     ),
    112     cast_to=Assistant,
    113 )

File ~/opt/miniconda3/envs/llm/lib/python3.9/site-packages/openai/_base_client.py:1055, in SyncAPIClient.post(self, path, cast_to, body, options, files, stream, stream_cls)
   1041 def post(
   1042     self,
   1043     path: str,
   (...)
   1050     stream_cls: type[_StreamT] | None = None,
   1051 ) -> ResponseT | _StreamT:
   1052     opts = FinalRequestOptions.construct(
   1053         method="post", url=path, json_data=body, files=to_httpx_files(files), **options
   1054     )
-> 1055     return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))

File ~/opt/miniconda3/envs/llm/lib/python3.9/site-packages/openai/_base_client.py:834, in SyncAPIClient.request(self, cast_to, options, remaining_retries, stream, stream_cls)
    825 def request(
    826     self,
    827     cast_to: Type[ResponseT],
   (...)
    832     stream_cls: type[_StreamT] | None = None,
    833 ) -> ResponseT | _StreamT:
--> 834     return self._request(
    835         cast_to=cast_to,
    836         options=options,
    837         stream=stream,
    838         stream_cls=stream_cls,
    839         remaining_retries=remaining_retries,
    840     )

File ~/opt/miniconda3/envs/llm/lib/python3.9/site-packages/openai/_base_client.py:877, in SyncAPIClient._request(self, cast_to, options, remaining_retries, stream, stream_cls)
    874     # If the response is streamed then we need to explicitly read the response
    875     # to completion before attempting to access the response text.
    876     err.response.read()
--> 877     raise self._make_status_error_from_response(err.response) from None
    878 except httpx.TimeoutException as err:
    879     if retries > 0:

BadRequestError: Error code: 400 - {'error': {'message': "8 validation errors for Request\nbody -> tools -> 0 -> type\n  unexpected value; permitted: <ToolTypeParam.CODE_INTERPRETER: 'code_interpreter'> (type=value_error.const; given=function; permitted=(<ToolTypeParam.CODE_INTERPRETER: 'code_interpreter'>,))\nbody -> tools -> 0 -> type\n  unexpected value; permitted: <ToolTypeParam.RETRIEVAL: 'retrieval'> (type=value_error.const; given=function; permitted=(<ToolTypeParam.RETRIEVAL: 'retrieval'>,))\nbody -> tools -> 0 -> function -> name\n  field required (type=value_error.missing)\nbody -> tools -> 0 -> function -> parameters\n  field required (type=value_error.missing)\nbody -> tools -> 1 -> type\n  unexpected value; permitted: <ToolTypeParam.CODE_INTERPRETER: 'code_interpreter'> (type=value_error.const; given=function; permitted=(<ToolTypeParam.CODE_INTERPRETER: 'code_interpreter'>,))\nbody -> tools -> 1 -> type\n  unexpected value; permitted: <ToolTypeParam.RETRIEVAL: 'retrieval'> (type=value_error.const; given=function; permitted=(<ToolTypeParam.RETRIEVAL: 'retrieval'>,))\nbody -> tools -> 1 -> function -> name\n  field required (type=value_error.missing)\nbody -> tools -> 1 -> function -> parameters\n  field required (type=value_error.missing)", 'type': 'invalid_request_error', 'param': None, 'code': None}}
```

**Comment by carterrees-entrata** [comment]
Update: looks like this might be relevant. I'll follow the solution here and report back: Issue [8853](https://github.com/run-llama/llama_index/issues/8853)

**Comment by carterrees-entrata** [comment]
This is indeed fixed based on the ticket in #8853.
